### ISSUE TITLE
failure in TestServerCacheRefresh

### ISSUE THREAD

**Issue by rade** [weaveworks/weave#892]
https://travis-ci.org/weaveworks/weave/builds/66364210
````
INFO: 2015/06/11 11:20:58.893469 TestServerCacheRefresh starting
INFO: 2015/06/11 11:20:58.894140 [zonedb] Using mDNS on all interfaces
INFO: 2015/06/11 11:20:58.894146 [zonedb] Starting 4 background name updaters
INFO: 2015/06/11 11:20:58.894152 [zonedb] Using mDNS on all interfaces
INFO: 2015/06/11 11:20:58.894154 [zonedb] Starting 4 background name updaters
INFO: 2015/06/11 11:20:58.894314 [dns] Upstream server(s): &{Servers:[10.1.0.2] Search:[travis] Port:53 Ndots:1 Timeout:5 Attempts:2}
INFO: 2015/06/11 11:20:58.894319 [dns] Cache: 1024 entries
INFO: 2015/06/11 11:20:58.894416 [dns] Listening for DNS on :58358 (UDP)
INFO: 2015/06/11 11:20:58.894416 [dns] Listening for DNS on :58358 (TCP)
INFO: 2015/06/11 11:20:58.995111 [dns msgid 32172] No results for type A query for 'second.weave.local.' [caching no-local]
INFO: 2015/06/11 11:20:58.996618 [dns] Server exiting...
--- FAIL: TestServerCacheRefresh-2 (0.10s)
	mocks_test.go:458: Response from '127.0.0.1:58358':
		;; opcode: QUERY, status: NOERROR, id: 61989
		;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
		
		;; QUESTION SECTION:
		;first.weave.local.	IN	 A
		
		;; ANSWER SECTION:
		first.weave.local.	30	IN	A	10.2.2.1
		
	mocks_test.go:458: Response from '127.0.0.1:58358':
		;; opcode: QUERY, status: NXDOMAIN, id: 32172
		;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0
		
		;; QUESTION SECTION:
		;second.weave.local.	IN	 A
		
	util.go:134: Expected nil pointer for value in the cache: after asking for first.weave.local.
		;; opcode: QUERY, status: NOERROR, id: 61989
		;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
		
		;; QUESTION SECTION:
		;first.weave.local.	IN	 A
		
		;; ANSWER SECTION:
		first.weave.local.	18	IN	A	10.2.2.1
		 but got a "*dns.Msg": ";; opcode: QUERY, status: NOERROR, id: 61989
		;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
		
		;; QUESTION SECTION:
		;first.weave.local.	IN	 A
		
		;; ANSWER SECTION:
		first.weave.local.	18	IN	A	10.2.2.1
		"
		goroutine 227 [running]:
		github.com/weaveworks/weave/testing.stackTrace(0x81c200, 0x0, 0x0)
			/home/travis/gopath/src/github.com/weaveworks/weave/testing/util.go:143 +0x88
		github.com/weaveworks/weave/testing.StackTrace(0x0, 0x0)
			/home/travis/gopath/src/github.com/weaveworks/weave/testing/util.go:138 +0x35
		github.com/weaveworks/weave/testing.Fatalf(0xc2081601b0, 0x8b2990, 0x30, 0xc2081c0b88, 0x3, 0x3)
			/home/travis/gopath/src/github.com/weaveworks/weave/testing/util.go:134 +0x27
		github.com/weaveworks/weave/testing.AssertNil(0xc2081601b0, 0x81c260, 0xc208160360, 0xc2080865a0, 0x115)
			/home/travis/gopath/src/github.com/weaveworks/weave/testing/util.go:26 +0x288
		github.com/weaveworks/weave/nameserver.assertNotInCache(0xc2081601b0, 0x2b0f45f44558, 0xc2080a80a0, 0xc2080e21b0, 0xc208152600, 0x23)
			/home/travis/gopath/src/github.com/weaveworks/weave/nameserver/mocks_test.go:497 +0x212
		github.com/weaveworks/weave/nameserver.TestServerCacheRefresh(0xc2081601b0)
			/home/travis/gopath/src/github.com/weaveworks/weave/nameserver/server_cache_test.go:323 +0xda7
		testing.tRunner(0xc2081601b0, 0xa21d78)
			/usr/local/go/src/testing/testing.go:447 +0xbf
		created by testing.RunTests
			/usr/local/go/src/testing/testing.go:555 +0xa8b
````

**Comment by inercia** [comment]
Failure rate seems to be reduced with benbjohnson/clock#5, but it still fails in some occasions...

**Comment by rade** [comment]
@inercia can this be closed? @bboreham believes this issue has been fixed in #917.

**Comment by inercia** [comment]
After running the test 1200 times, this failure did not appear... I guess we can close it...
